src/app/components/keywords/index.tsx

```diff
@@ -61,25 +67,35 @@ export const KeywordsEditor: React.FC<KeywordsEditorProps> = (props) => {
     }
   }, [keywordsDomContainer]);
 
-  const renderKeywords = (keywords: EditorState[]) => {
-    return keywords.map((keywordEditorState, index) => {
+  const renderKeywords = (keywords: Keyword[]): React.ReactNode[] => {
+    return keywords.map((keyword, index) => {
       return (
-        <Keyword
+        <KeywordSection
           key={index}
           isActive={isActiveKeyword(name, index, activeKeywordPath)}
-          onChange={onChange.bind(null, name, index)}
-          editorState={keywordEditorState}
-          onDelete={onDelete.bind(null, name, index)}
-          onFocus={handleFocus.bind(null, index)}
+          onChange={onChange.bind(null, name)}
+          keyword={keyword}
+          onDelete={onDelete.bind(null, name)}
+          onFocus={handleFocus.bind(null)}
           onBlur={onBlur}
         />
       );
     });
   };
 
-  const handleNewKeywordFocus = useCallback(() => {
-    onFocus(name, undefined, true);
-  }, [name, onFocus]);
+  const handleNewKeywordFocus = useCallback(
+    (state: EditorState) => {
+      onFocus(state, `keywordGroups.${name}.keywords.newKeyword.content`);
+    },
+    [name, onFocus]
+  );
+
+  const handleAddKeyword = useCallback(
```

Comment:
Testing locally, there is an issue if I add a new keyword and then press undo.
1. Add a new keyword, e.g. 'test'.
2. Press Undo.
Results in...
![Screenshot 2021-01-12 at 10 12 29](https://user-images.githubusercontent.com/52923935/104300680-b315c580-54be-11eb-9fe6-8b7bec6bb41d.png)

Reply:
So this is the same behaviour we have in staging currently. Shall we ask Fred and Nick to test and raise a bug if necessary?

Reply:
No it isn't, just checked staging myself. But I can't recreate the issue on the latest version of this PR so I assume that it's now fixed.

Reply:
I have found another issue with keywords though...
1. Add a new keyword, e.g. Test.
2. Use undo on the toolbar to undo the change, note this doesn't remove the keyword, but instead puts it back into edit mode with all the text.
3. Use redo on the toolbar, hits an exception.

Note, this isn't reproducible in staging.

Reply:
Fixed in latest commit